packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m

```diff
@@ -44,7 +44,21 @@ - (dispatch_queue_t)methodQueue {
         rejecter:
         (RCTPromiseRejectBlock) reject) {
     @try {
-      [FIRAnalytics logEventWithName:name parameters:params];
+      if (params[kFIRParameterItems]) {
+        NSMutableDictionary *newParams = [params mutableCopy];
+        NSMutableArray *newItems = [NSMutableArray array];
+        [(NSArray *)params[kFIRParameterItems] enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+          NSMutableDictionary *item = [obj mutableCopy];
+          if (item[kFIRParameterQuantity]) {
+            item[kFIRParameterQuantity] = @([item[kFIRParameterQuantity] integerValue]);
```

Comment:
Just to make sure I understand, this is effectively a type conversion (from number which I guess goes to float instead of integer?) in native code because you need it to be an native integer (as discovered by testing) for it to show up in the event on the analytics console?

Reply:
Exactly. `item[kFIRParameterQuantity]` is of type `NSNumber` and its `integerValue` method gives back a pure whole number. Then the value is wrapped with `@()`, which is a shorthand for creating an NSNumber, equivalent would be:
`[NSNumber numberWithInteger:[item[kFIRParameterQuantity] integerValue]]`.

Reply:
Great! My Obj-C is weak but apparently (finally!!!) getting stronger if I got that correct :-). Thank you for confirming